swagger/util.go

```diff
@@ -12,20 +12,27 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-//
 package swagger
 
 import (
 	"fmt"
+	"net/url"
 	"path/filepath"
+	"regexp"
 	"strings"
 )
 
 // UsePackageName can be set to true to add package prefix of generated definition names
 var UsePackageName = false
 
+// StripPackagePrefixes can be set to remove leading strings from long package names, eg github.com/some-ORG/
+// So github.com/some-ORG/repo/types.Pet becomes repo/types.Pet
+var StripPackagePrefixes []string
+
+var genericTypeRegex = regexp.MustCompile(`(?P<type>[a-zA-Z0-9_\.\-\/]*)\[(?P<typeParams>([a-zA-Z0-9_\.\-\/]+(, ?)?)+)\]`)
```

Comment:
This can be simplified to
`(?P<type>[\w.\-/]+)\[(?P<typeParams>([\w.\-/]+(, ?)?)+)\]`

Actually, since that's already quite a bit more tolerant than the language really allows, you could get away with
`(?P<type>[\w.\-/]+)\[(?P<typeParams>[\w.\-/,\s]+)\]`

Reply:
I didn't know that you can have `/` without escapes. And `\w` is a nice shorthand. Thanks.